mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -118,6 +117,31 @@ Type Conv2dGenerator::getDataType(OpBuilder &builder) const {
   return dataType;
 }
 
+Optional<miopen::ConvOpType>
+Conv2dGenerator::getOpForName(const StringRef name) {
+  if (name == "conv2d") {
+    return miopen::Conv2DOpType;
+  }
+  if (name == "conv2d_bwd_data") {
+    return miopen::Conv2DBwdDataOpType;
+  }
+  if (name == "conv2d_bwd_weight") {
+    return miopen::Conv2DBwdWeightOpType;
+  }
+  return llvm::None;
+}
+
+const char *Conv2dGenerator::getNameForOp(const miopen::ConvOpType op) {
+  switch (op) {
+  case miopen::Conv2DOpType:
+    return "conv2d";
+  case miopen::Conv2DBwdDataOpType:
+    return "conv2d_bwd_data";
+  case miopen::Conv2DBwdWeightOpType:
+    return "conv2d_bwd_weight";
+  }
+}
+
```

Comment:
I think it is a better idea for those two convenient functions to stay close to the `miopen::ConvOpType`. I believe we did this sort of translation in `LowerMIOpenOps.h` as well. But for the scope of this review, just add the convenience function to where the type is should be enough.

Reply:
And fixed.